railties/lib/rails/engine/configuration.rb

```diff
@@ -89,6 +89,11 @@ def autoload_once_paths
       def autoload_paths
         @autoload_paths ||= paths.autoload_paths
       end
+
+      # Called after application routes have been loaded.
+      def after_routes_loaded(&block)
```

Comment:
We should move to the Railtie configuration since they can also configure some behavior to happen after route reloading. WE should also document this method in this guide like we do with `after_initialize`. https://github.com/rails/rails/blob/main/guides/source/configuring.md